### PR TITLE
Feat: edit product

### PR DESCRIPTION
**Pull request by MathiasDuus**

### CODE REVIEW

**Review by Setanna — CHANGES_REQUESTED** [review]

File: lib/misc/api/api_product.dart [hunk 1]
```diff
@@ -156,3 +156,83 @@ Future<Map<String, dynamic>> fetchProduct(int id) async {
...
+      'category_id': categoryId,
+      'content': content,
+      'unit_id': unitId,
+      'barcode': barcode,
+      'locations': locations,
+    }),
+  );
+
+  // Check if the request was successful
+  return response.statusCode == 200;
```
[the remark this answers is not in the document]
Comment (MathiasDuus) [reply lib/misc/api/api_product.dart]: done

File: lib/modal/add_product_modal.dart [hunk 2]
```diff
@@ -73,8 +73,8 @@ Future addProductModal(
             ),
             DropdownSearch(
               asyncItems: (String filter) async {
-                final categories = await api.fetchUnits();
-                return categories;
+                final units = await api.fetchUnits();
+                return units;
```
[the remark this answers is not in the document]
Comment (MathiasDuus) [reply lib/modal/add_product_modal.dart:77]: done

File: lib/modal/edit_product_modal.dart [hunk 3]
```diff
@@ -0,0 +1,293 @@
...
+                    isFormValid = formKey.currentState!.validate();
+                  });
+                }
+              },
+              child: Column(
+                children: [
+                  DropdownSearch(
+                    asyncItems: (String filter) async {
+                      final locations = await api.fetchLocations();
+                      return locations;
```
[the remark this answers is not in the document]
Comment (MathiasDuus) [reply lib/modal/edit_product_modal.dart:61]: done

**Comment by MathiasDuus** [comment]
[GUI - Edit product](https://trello.com/c/w0swbIzT/29-gui-edit-product)

**Review by Setanna — CHANGES_REQUESTED** [review]
Jane Doe (Moderator) can't press the update button, it is disabled in the modal.

**Comment by MathiasDuus** [comment]
It sems BaseItem did not have an isEmpty getter.
Somehow when updating the product it just disapeared from the DB

File: lib/modal/edit_product_modal.dart [hunk 4]
```diff
@@ -0,0 +1,358 @@
...
+                      if (value == null || value.isEmpty) {
+                        return '';
+                      }
+
+                      if (errors.containsPartialKey('unit')) {
+                        return errors[errors.getValue('unit')][0];
+                      }
+                      return null;
+                    },
+                  ),
```
Comment (Setanna) [inline lib/modal/edit_product_modal.dart:216]: Shows unit abbreviation on load, but shows abbreviation name in dropdown and on change

File: lib/modal/edit_product_modal.dart [hunk 5]
```diff
@@ -0,0 +1,358 @@
...
+                      }
+
+                      if (errors.containsPartialKey('shelf')) {
+                        return errors[errors.getValue('shelf')][0];
+                      }
+                      return null;
+                    },
+                  ),
+                ],
+              ),
```
Comment (Setanna) [inline lib/modal/edit_product_modal.dart:276]: Low priority. Could change stock and shelf amount based on location chosen

**Review by Setanna — CHANGES_REQUESTED** [review]

File: lib/modal/edit_product_modal.dart (hunk 4 above)
Comment (MathiasDuus) [reply lib/modal/edit_product_modal.dart]: Fixed, name is now in the pre-select also.

File: lib/modal/edit_product_modal.dart (hunk 5 above)
Comment (MathiasDuus) [reply lib/modal/edit_product_modal.dart:276]: No, I will make a seperate story for this :)

**Review by Setanna — APPROVED** [review]

**Pull request merged**